active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,220 @@
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+## Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+Number box is a numerics only text control with support for input formatting, validation, basic calculation, and increment stepping.
+
+**Important APIs:** [NumberBox class](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.numberbox)
+
+## Is this the right control? 
+
+Use a **NumberBox** control to recieve and display formatted numeric values such as currency, IP addresses, and telephone numbers. A **NumberBox** can also be enabled with basic calculator support to compute multiplication, division, addition, and subtraction across order of operations on entered equations. 
+
+## Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox"/>
+```
+
+### Add placeholder text and a header
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+### Enable calculation support
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
+    PlaceholderText="A + B * C" 
+    AcceptsCalculation="True" />
+```
+
+### Add formatting
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for IP addresss" 
+    Header="Enter equation:" 
+    PlaceholderText="000.000.000.000"
+    FormatType="IPAddress" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for international telephone" 
+    Header="Enter equation:" 
+    PlaceholderText="0 (123) 456-7890"
+    FormatType="InternationalTelephone" />
+```
+
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for custom input" 
+    Header="Enter equation:" 
+    PlaceholderText="00 ft 00 in"
+    CustomFormatString="## ft ## in"
+    StripLeadingZeros="False" />
+```
+
+### Add increment and decrement stepping
+
+XAML
+```XAML
+<TextBlock AutomationProperties.Name="NumberBox for items in order" 
+    Header="Order:" 
+    Text="2"
+    StepSize="2"
```

Comment:
`StepSize="2"` What is this property? Where did it come from? Why is it not documented elsewhere?

Reply:
I would assume the StepSize property is the equivalent of the **StepFrequency** that is used on the Slider control.

I guess it would control how much the value will increment by when pressing the Up and Down Arrow keys or the Spin Buttons?

Reply:
guesses and assumptions don't matter--even if they're right. Having this documented in the spec does.
The purpose of the spec is to clearly document everything that the control should do. This matters when it comes time to implement the functionality, so the developer knows what to implement. It matters during testing, so that it's possible to verify that the control correctly does everything it is supposed to. And it matters in the future as the spec is used to produce documentation so the consumers of the control know how it works.

Having an undocumented property in an example will only lead to questions and uncertainty. The purpose of reviewing the spec is to make sure there isn't any.

Where a question is asked in a review comment it is not beneficial to further discuss this in comments, unless the question is unclear and needs clarification. The correct response should be to update the document so that no-one else would ask that question upon reading the document for the first time in the future. The spec needs to standalone and not rely on people also reading comments to understand it.

Reply:
> I guess it would control how much the value will increment or decrement by when pressing the Up and Down Arrow keys

Will arrow keys update the value shown inside the NumberBox? Left/Right arrows have to move focus within the NumberBox. I suppose Up/Down arrows can change the value, but that's not typical for Xaml controls. Slider is the one example I can think of.

Reply:
In the NumericUpDown in WinRT XAML Toolkit they do increment/decrement the value.

Reply:
Same in the Visual Studio designer.

Reply:
> guesses and assumptions don't matter--even if they're right. Having this documented in the spec does.
> The purpose of the spec is to clearly document everything that the control should do. This matters when it comes time to implement the functionality, so the developer knows what to implement. It matters during testing, so that it's possible to verify that the control correctly does everything it is supposed to. And it matters in the future as the spec is used to produce documentation so the consumers of the control know how it works.
> 
> Having an undocumented property in an example will only lead to questions and uncertainty. The purpose of reviewing the spec is to make sure there isn't any.
> 
> Where a question is asked in a review comment it is not beneficial to further discuss this in comments, unless the question is unclear and needs clarification. The correct response should be to update the document so that no-one else would ask that question upon reading the document for the first time in the future. The spec needs to standalone and not rely on people also reading comments to understand it.

Agreed - you have no resistance here! But because we have committed fully to open source - that includes starting our spec drafts open source as well. My normally drafting routine has been usurped to a degree by my efforts to respond to everyone here and prioritize resolutions on large and popular topics. Thanks for your patience in getting to this one. : )

@mdtauk's assessment aligned with my intention and I have since updated the property to match the name of the same property on Slider and added its description to the API Notes section which will later be turned into the MSDN API docs. I will come back by this example and add a comprehensive description as soon as I have finished corralling a majority of the API - it makes writing the examples easier when I can plan how to order them and where I can combine examples.

Reply:
> guesses and assumptions don't matter--even if they're right. Having this documented in the spec does.
> The purpose of the spec is to clearly document everything that the control should do. This matters when it comes time to implement the functionality, so the developer knows what to implement. It matters during testing, so that it's possible to verify that the control correctly does everything it is supposed to. And it matters in the future as the spec is used to produce documentation so the consumers of the control know how it works.
> 
> Having an undocumented property in an example will only lead to questions and uncertainty. The purpose of reviewing the spec is to make sure there isn't any.
> 
> Where a question is asked in a review comment it is not beneficial to further discuss this in comments, unless the question is unclear and needs clarification. The correct response should be to update the document so that no-one else would ask that question upon reading the document for the first time in the future. The spec needs to standalone and not rely on people also reading comments to understand it.

@mrlacey, Agreed - you have no resistance here! But because we have committed fully to open source - that includes starting our spec drafts open source as well. My normally drafting routine has been usurped to a degree by my efforts to respond to everyone here and prioritize resolutions on large and popular topics. Thanks for your patience in getting to this one. : )

@mdtauk's assessment aligned with my intention and I have since updated the property to match the name of the same property on Slider and added its description to the API Notes section which will later be turned into the MSDN API docs. I will come back by this example and add a comprehensive description as soon as I have finished corralling a majority of the API - it makes writing the examples easier when I can plan how to order them and where I can combine examples.